spec/test_track_spec.rb

```diff
@@ -275,5 +277,25 @@
         end
       end
     end
+
+    context 'when set via full assets:precompile command' do
+      around do |example|
+        FileUtils.remove_dir('testtrack', true)
+        example.run
+        Rake::Task['app:assets:clobber'].invoke
+        FileUtils.remove_dir('testtrack', true)
+      end
+
+      let(:asset_precompile_success) do
+        system({ 'RAILS_ENV' => 'production' }, 'bundle exec rake app:assets:precompile')
+      end
+
+      it 'does not raise an error' do
+        expect { asset_precompile_success }
+          .to change { File.exist?('testtrack/build_timestamp') }
+          .from(false).to(true)
+        expect(asset_precompile_success).to eq true
+      end
+    end
```

Comment:
I think there's a more surgical way of doing this. If you `cd` into the dummy app, then you can avoid mucking with the Rakefile and all the other rake loading things you had to do. It also means that you are operating with that app's `testtrack` directory, rather than a fake `testtrack` directory at the root of the project. With just this change plus turning off uglifier if the dummy app's prod config, I was able to get all the tests passing, and it feels closer to a real world scenario.

```suggestion
    context 'when set via full assets:precompile command' do
      around do |example|
        Dir.chdir('spec/dummy') do
          File.delete('testtrack/build_timestamp') if File.exist?('testtrack/build_timestamp')
          example.run
          system({ 'RAILS_ENV' => 'production' }, 'bundle exec rake assets:clobber')
          File.delete('testtrack/build_timestamp') if File.exist?('testtrack/build_timestamp')
        end
      end

      let(:asset_precompile_success) do
        system({ 'RAILS_ENV' => 'production' }, 'bundle exec rake assets:precompile')
      end

      it 'does not raise an error' do
        expect { asset_precompile_success }
          .to change { File.exist?('testtrack/build_timestamp') }
          .from(false).to(true)
        expect(asset_precompile_success).to eq true
      end
    end
```

Reply:
Okay! Happy to do that if it feels less invasive!

I've been taking a strategy with some gems where I enable full functionality at the root, including a `rackup` that lets me actually boot the dummy app and click around a test server. But I can understand how that would not be the goal of this PR. 🙂

Reply:
And to clarify, would you prefer I leave the prod config intact and _just_ turn off uglifier?